add instructions to use template feature, update docs (#57)
carpentries/training-template: README.md

    (If you do not have an account, you can quickly create one for free.)
    You must be logged in for the remaining steps to work.

2.  On this page (<https://github.com/carpentries/training-template>),
    click on the green "Use this template" button (top right)

    ![screenshot of this repository's GitHub page with an arrow pointing to the the 'use this template' button on the top left](fig/select-github-use-template.png?raw=true)

3.  Select the owner for your new repository.
    (This will probably be you, but may instead be an organization you belong to.)

4.  Choose a name for your workshop website repository.
    This name should have the form `YYYY-MM-DD-ttt-site`,
    e.g., `2016-12-01-ttt-oomza`,
    where `YYYY-MM-DD` is the start date of the workshop.

>     for online workshops, choose `online` as `site`.

Reviewer: ```suggestion
   For online workshops, choose `online` as `site`.
```

>     In any case, the Instructor Training Team will tell you

Reviewer: ```suggestion
    In most cases, the Instructor Training Team will tell you
```

    git push origin gh-pages
    ~~~

7.  When you are done editing,
    go to the GitHub Pages URL for your workshop and preview your changes.
    In the example above, this is `https://gvwilson.github.io/2016-12-01-ttt-oomza`.
    The finished page should look [something like this](fig/completed-page.png?raw=true).

8.  Optional: you can now change the `README.md` file in your website's repository, which contains these instructions, so that it contains a short description of your workshop and a link to the training website.

9.  Optional: Add a link to your workshop website on the repository main page in the description/website section (look for the `Edit` button on the right to add).

**Note:**
please do all of your work in your repository's `gh-pages` branch,
since [GitHub automatically publishes that as a website][github-project-pages].

> ## Checking Your Changes Locally

Reviewer: ```suggestion
## (Optional) Checking Your Changes Locally
```
(because I never do this and I think that's ok :)
